src/main/java/org/ozwillo/dcexporter/service/DcModelMappingService.java

```diff
@@ -62,12 +64,32 @@ public DcModelMapping getById(String id) {
         dcModelMapping.setDeleted(false);
 
         dcModelMapping = dcModelMappingRepository.save(dcModelMapping);
+        SynchronizerAuditLog newAuditLog = new SynchronizerAuditLog(dcModelMapping.getType(), SynchronizerStatus.PENDING, null,  DateTime.now());
+        synchronizerAuditLogRepository.save(newAuditLog);
         return Either.right(dcModelMapping);
     }
 
     public Either<String, DcModelMapping> edit(DcModelMapping dcModelMapping) {
 
-        if (dcModelMappingRepository.findByDcId(dcModelMapping.getDcId()) == null) return Either.left("dataset.notif.not_exist");
+        DcModelMapping oldDcModelMapping = dcModelMappingRepository.findByDcId(dcModelMapping.getDcId());
+
+        if(oldDcModelMapping == null) return Either.left("dataset.notif.not_exist");
+
+        SynchronizerAuditLog newAuditLog = new SynchronizerAuditLog(dcModelMapping.getType(), SynchronizerStatus.MODIFIED, null,  DateTime.now());
+        synchronizerAuditLogRepository.save(newAuditLog);
+        dcModelMappingRepository.save(dcModelMapping);
```

Comment:
It's safer to save the changes before processing, and again afterwards to add the new values ( Ckan Package Id, Ckan Resource Id .. )